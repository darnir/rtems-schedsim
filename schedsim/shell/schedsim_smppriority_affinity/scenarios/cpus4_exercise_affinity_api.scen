#
#  Verify that affinity APIs work with the right values
#
#  This scenario does **NOT** change the affinity of any thread
#  which ever executes.
#
rtems_init 4
task_create smp1 3
task_create smp2 6
task_create smp3 7
task_create smp4 8
task_create smp5 10
dispatch
# GOAL: Tasks 0x0a010001 - 0x0a010004 as heir and executing
# GOAL:    on CPUs 1 - 4 respectively
cpus

# GOAL: These should work OK and change the affinity
task_get_affinity smp5
task_set_affinity smp5 0x01
task_get_affinity smp5
task_set_affinity smp5 0x02
task_get_affinity smp5
task_set_affinity smp5 0x04
task_get_affinity smp5
task_set_affinity smp5 0x08
task_get_affinity smp5

# GOAL This should fail.
task_set_affinity smp5 0x10
task_get_affinity smp5

