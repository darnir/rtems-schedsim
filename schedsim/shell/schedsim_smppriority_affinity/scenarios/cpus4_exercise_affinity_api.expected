Script File               : scenarios/cpus4_exercise_affinity_api.scen
verbose                   : 1
==> 1: #
==> 2: #  Verify that affinity APIs work with the right values
==> 3: #
==> 4: rtems_init 4
  CPU 0: Thread Heir 0x09010001 priority=255
  CPU 0: Thread Executing 0x09010001 priority=255
  CPU 1: Thread Heir 0x09010002 priority=255
  CPU 1: Thread Executing 0x09010002 priority=255
  CPU 2: Thread Heir 0x09010003 priority=255
  CPU 2: Thread Executing 0x09010003 priority=255
  CPU 3: Thread Heir 0x09010004 priority=255
  CPU 3: Thread Executing 0x09010004 priority=255
==> 5: task_create smp1 3
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=3
Task (smp1) starting: id=0x0a010001, priority=3
  CPU 3: Thread Heir 0x0a010001 priority=3
  CPU 3: Thread Executing 0x0a010001 priority=3
==> 6: task_create smp2 6
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=6
Task (smp2) starting: id=0x0a010002, priority=6
  CPU 2: Thread Heir 0x0a010002 priority=6
  CPU 2: Thread Executing 0x0a010002 priority=6
==> 7: task_create smp3 7
Creating task with:  Preemption: yes Timeslicing: no
Task (smp3) created: id=0x0a010003, priority=7
Task (smp3) starting: id=0x0a010003, priority=7
  CPU 1: Thread Heir 0x0a010003 priority=7
  CPU 1: Thread Executing 0x0a010003 priority=7
==> 8: task_create smp4 8
Creating task with:  Preemption: yes Timeslicing: no
Task (smp4) created: id=0x0a010004, priority=8
Task (smp4) starting: id=0x0a010004, priority=8
  CPU 0: Thread Heir 0x0a010004 priority=8
  CPU 0: Thread Executing 0x0a010004 priority=8
==> 9: task_create smp5 10
Creating task with:  Preemption: yes Timeslicing: no
Task (smp5) created: id=0x0a010005, priority=10
Task (smp5) starting: id=0x0a010005, priority=10
==> 10: dispatch
==> 11: # GOAL: Tasks 0x0a010001 - 0x0a010004 as heir and executing
==> 12: # GOAL:    on CPUs 1 - 4 respectively
==> 13: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  8 / 0x0a010004 @  8          false
  CPU 1: 0x0a010003 @  7 / 0x0a010003 @  7          false
  CPU 2: 0x0a010002 @  6 / 0x0a010002 @  6          false
  CPU 3: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
==> 14: 
==> 15: # GOAL: These should work OK and change the affinity
==> 16: task_get_affinity smp5
Task (0x0a010005) Get affinity=0x0000000f
==> 17: task_set_affinity smp5 0x01
Task (0x0a010005) Set affinity=0x00000001
==> 18: task_get_affinity smp5
Task (0x0a010005) Get affinity=0x00000001
==> 19: task_set_affinity smp5 0x02
Task (0x0a010005) Set affinity=0x00000002
==> 20: task_get_affinity smp5
Task (0x0a010005) Get affinity=0x00000002
==> 21: task_set_affinity smp5 0x04
Task (0x0a010005) Set affinity=0x00000004
==> 22: task_get_affinity smp5
Task (0x0a010005) Get affinity=0x00000004
==> 23: task_set_affinity smp5 0x08
Task (0x0a010005) Set affinity=0x00000008
==> 24: task_get_affinity smp5
Task (0x0a010005) Get affinity=0x00000008
==> 25: 
==> 26: # GOAL This should fail.
==> 27: task_set_affinity smp5 0x10
==> 28: task_get_affinity smp5
Task (0x0a010005) Get affinity=0x00000008
==> 29: 
